### PR TITLE
Add Spacing for Nonempty Brackets Formatting Option

### PR DESCRIPTION
**Pull request by ToddThomson**
User formatting option to add single space after opening and before closing nonempty brackets. Addresses issue #2795
Testing 
Fourslash test: formattingOptionsChange.ts modified to test new formatting option.

Notes
All tests passing

There is a test named formattingOnClosingBracket.ts . The filename should be formattingOnClosingBrace.ts

### CODE REVIEW

File: src/services/formatting/rules.ts [hunk 1]
```diff
@@ -451,6 +458,13 @@ namespace ts.formatting {
             this.NoSpaceAfterOpenParen = new Rule(RuleDescriptor.create3(SyntaxKind.OpenParenToken, Shared.TokenRange.Any), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext), RuleAction.Delete));
             this.NoSpaceBeforeCloseParen = new Rule(RuleDescriptor.create2(Shared.TokenRange.Any, SyntaxKind.CloseParenToken), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext), RuleAction.Delete));
 
+            // Insert space after opening and before closing nonempty brackets
+            this.SpaceAfterOpenBracket = new Rule( RuleDescriptor.create3( SyntaxKind.OpenBracketToken, Shared.TokenRange.Any ), RuleOperation.create2( new RuleOperationContext( Rules.IsSameLineTokenContext ), RuleAction.Space ) );
```
Comment (mhegazy) [inline src/services/formatting/rules.ts]: :D i would appreciate it if you do not have spaces after and/or before the parens for this code change.

File: tests/cases/fourslash/formattingOptionsChange.ts [hunk 2]
```diff
@@ -17,6 +18,7 @@ runTest("InsertSpaceBeforeAndAfterBinaryOperators", "1 + 2 - 3", "1+2-3");
 runTest("InsertSpaceAfterKeywordsInControlFlowStatements", "if (true) { }", "if(true) { }");
 runTest("InsertSpaceAfterFunctionKeywordForAnonymousFunctions", "(function () { })", "(function() { })");
 runTest("InsertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis", "    ( 1 )", "    (1)");
+runTest( "InsertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets", "[ 1 ]", "[1]" );
```
Comment (mhegazy) [inline tests/cases/fourslash/formattingOptionsChange.ts]: can we add a test for something with empty brackets, and brackets with only commas.

**Comment by mhegazy** [comment]
thanks @ToddThomson, just a couple of nit comments. otherwise :+1:

**Comment by ToddThomson** [comment]
This is the first time I've gone through the process on using GitHub to submit a PR. The only snag I hit was not knowing that I had to fork the typescript repo to my personal github account before making changes to the source.

**Comment by mhegazy** [comment]
glad that was smooth.. now we can expect more contributions :)

may be you would want to add a space after `{` and before `}` in module imports/exports..

**Comment by ToddThomson** [comment]
@mhegazy  You wrote :D i would appreciate it if you do not have spaces after and/or before the parens for this code change.

No worries! I do like my spaces :smile: I'll make the changes as you suggest.

File: tests/cases/fourslash/formattingOptionsChange.ts (hunk 2 above)
Comment (ToddThomson) [inline tests/cases/fourslash/formattingOptionsChange.ts]: @mhegazy Your request to add a test for brackets with only commas is interesting. With the current formatting options, `[,]` becomes `[, ]`, if "insert space after comma delimiter" is true. With this PR, the result would be: `[ , ]` .

**Comment by mhegazy** [comment]
Thanks!

**Pull request merged**